### DISCUSSION TITLE
Mass accept suggestions as translations

### DISCUSSION THREAD

**Discussion by Glandos** [WeblateOrg/weblate#8554] [Q&A] [answered]
Hi,

An anonymous user forgot to log in before translating 460 strings, and thus submitted them as suggestions.
However, before going in the tedious process of clicking 460 times, is there a way to accept all unique suggestions as translations?
I'm opened for shell-fu solutions!

**Comment by nijel** [comment, marked as answer]
There is currently no interface for that (see https://github.com/WeblateOrg/weblate/issues/671).

You can do that from Python from `weblate shell`. Something like this should work (not tested):

```python
from weblate.trans.models import Suggestion
from weblate.auth.models import User
from django.http import HttpRequest

# Filter desired suggestions
suggestions = Suggestion.objects.filter(author__username="anonymous", unit__translation__component__project="PROJECT")

# Check if they look okayish
print(suggestions)

# Fake request
request = HttpRequest()
request.user = User.objects.get(username="YOUR_USERNAME")

# Accept all suggestions
for suggestion in suggestions:
    suggestion.accept(request)
